core/vm/evm.go

```diff
@@ -307,13 +311,17 @@ func (evm *EVM) Create(caller ContractRef, code []byte, gas uint64, value *big.I
 	if !evm.CanTransfer(evm.StateDB, caller.Address(), value) {
 		return nil, common.Address{}, gas, ErrInsufficientBalance
 	}
-
-	// Create a new account on the state
+	// Ensure there's no existing contract already at the designated address
 	nonce := evm.StateDB.GetNonce(caller.Address())
 	evm.StateDB.SetNonce(caller.Address(), nonce+1)
 
-	snapshot := evm.StateDB.Snapshot()
 	contractAddr = crypto.CreateAddress(caller.Address(), nonce)
+	contractHash := evm.StateDB.GetCodeHash(contractAddr)
+	if evm.StateDB.GetNonce(contractAddr) != 0 || (contractHash != (common.Hash{}) && contractHash != emptyCodeHash) {
```

Comment:
Why is this not `env.StateDB.Exist`?

Reply:
That results in a lot of test failures.

Reply:
I'm guessing it's some complexity with suicides within create.

Reply:
Does it work with `!env.StateDB.Empty(contractAddr)`? Trying to avoid the complicated condition here.

Reply:
Empty also requires balance == 0. That's not the case here, balance is allowed to be > 0.

Reply:
right